### PR TITLE
Feature/aip 779

### PR DESCRIPTION
**Pull request by fortunato**
This PR contains a scaffold of the primary exported component from this NPM package with its props interface geared towards allowing implementation with any backend.

See: https://datapunt.atlassian.net/browse/AIP-779

### CODE REVIEW

File: src/store/CustomProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+const CustomProvider: React.FC<Props> = ({ children }: Props) => {
+	return (
+		<Provider context={CustomContext} store={store}>
+			{children}
+		</Provider>
+	);
+};
+
+export default CustomProvider;
```
Comment (vdagraca) [inline src/store/CustomProvider.tsx:40]: Why do you need a provider for redux?

**Review by vdagraca — APPROVED** [review]

File: src/components/AddDocumentButton/AddDocumentButton.test.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	test('Renders a button using the given buttonText', () => {
+		renderComponent();
+		expect(screen.getByText('Upload', { selector: 'button' })).toBeInTheDocument();
+		expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+	});
+
+	test('Clicking button opens up wizard', () => {
+		renderComponent();
+		const button = screen.getByText('Upload', { selector: 'button' });
+		userEvent.click(button);
```
[the remark this answers is not in the document]
Comment (fortunato) [reply src/components/AddDocumentButton/AddDocumentButton.test.tsx:33]: userEvent is one fo the add-ons for react testing library: https://testing-library.com/docs/ecosystem-user-event/

I agree that in this instance there isn't any benefit in leveraging userEvent over fireEvent. In other cases userEvent should result in simpler test code. In other projects we tend to use it as well to simulate keyboard events and such.

[the remark this answers is not in the document]
Comment (fortunato) [reply src/components/AddDocumentButton/AddDocumentButton.test.tsx:33]: userEvent is one fo the add-ons for react testing library: https://testing-library.com/docs/ecosystem-user-event/

I agree that in this instance there isn't any benefit in leveraging userEvent over fireEvent. In other cases userEvent should result in simpler test code. In other projects we tend to use it as well to simulate keyboard events and such.

File: src/store/CustomProvider.tsx (hunk 1 above)
Comment (fortunato) [reply src/store/CustomProvider.tsx:40]: I read that to ensure we avoid conflicts between this store and a redux store already present in the application this is imported into, it is a good idea to create a provider with a custom context.

**Pull request merged**